### PR TITLE
XENOPS-882 - first version

### PR DESCRIPTION
**Pull request by wimfabri**

### CODE REVIEW

File: config/patroni/patroni-A/patronictl.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+{dcs_api: 'consul://consul:8500', namespace: /service/}
```
Comment (hvancamp) [inline config/patroni/patroni-A/patronictl.yaml:1]: file can be removed, no? it was also removed from docker-patroni

File: config/patroni/patroni-B/patronictl.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+{dcs_api: 'consul://consul:8500', namespace: /service/}
```
Comment (hvancamp) [inline config/patroni/patroni-B/patronictl.yaml:1]: file can be removed, no? it was also removed from docker-patroni

**Comment by hvancamp** [comment]
build.gradle file is missing?

File: src/main/compose/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    - 0.0.0.0::5432
+    - 0.0.0.0::8008
+    environment:
+    - PGDATA=/var/lib/postgresql/data/PGDATA
+    - PATRONI_NAME=postgresqlbravo
+    - PATRONI_SCOPE=alfdb
+    - PATRONI_POSTGRESQL_CONNECT_ADDRESS=postgresqlbravo:5432
+    - PATRONI_RESTAPI_CONNECT_ADDRESS=postgresqlbravo:8008
+
+  haproxy:
```
Comment (hvancamp) [inline src/main/compose/docker-compose.yml:32]: is this haproxy used for the build/tests?

File: src/main/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      curl \
+      gnupg \
+      apt-transport-https \
+      lsb-release \
+      make \
+      llvm \
+    && echo "deb https://packagecloud.io/timescale/timescaledb/debian/ $(lsb_release -c -s) main" > /etc/apt/sources.list.d/timescaledb.list \
+    && curl -L https://packagecloud.io/timescale/timescaledb/gpgkey | apt-key add - \
+    && apt-get update \
+    && apt-get install -y timescaledb-2-2.4.1-postgresql-13 \
```
Comment (hvancamp) [inline src/main/docker/Dockerfile:97]: should use the environment variable (timescaledb_version) defined in the build.gradle

File: src/main/docker/root/docker-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+if [[ $(stat -c %a "$PGDATA") != '700' ]]
+then
+  chmod 700 "$PGDATA"
+fi
+
+mkdir --parents /root/.config/patroni
+ln -s /config/patronictl.yaml /root/.config/patroni/patronictl.yaml
+
+/usr/bin/timescaledb-tune --quiet --yes --conf-path=$PGDATA --dry-run
```
Comment (hvancamp) [inline src/main/docker/root/docker-entrypoint.sh:21]: this command will only output the recommended pg settings but not apply them, see
https://docs.timescale.com/timescaledb/latest/how-to-guides/configuration/timescaledb-tune/

perhaps write the recommended pg settings into a new conf file in /config/patroni?

File: src/main/docker/root/post-init.sh [hunk 6]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
```
Comment (hvancamp) [inline src/main/docker/root/post-init.sh:1]: this file is already in the patroni base_image. so can be removed

File: src/main/docker/root/docker-entrypoint.sh (hunk 5 above)
Comment (wimfabri) [reply src/main/docker/root/docker-entrypoint.sh:21]: was a leftover from testing
timescaledb-tune needs a postgres config as input, so should run after patroni has created one
but patroni only uses these settings when initially creating the database I think

File: src/main/compose/docker-compose.yml (hunk 3 above)
Comment (wimfabri) [reply src/main/compose/docker-compose.yml:32]: this was copied from docker-patroni

**Comment by hvancamp** [comment]
can you also add git actions? to automatically build and push the image ...

**Review by hvancamp — APPROVED** [review]

**Comment by tgeens** [comment]
Nice work :+1:

**Pull request merged**